app/src/main/java/com/example/zingmp3phake/screen/detailplaylist/DetailPlaylistActivity.kt

```diff
@@ -1,17 +1,131 @@
+import android.view.View
+import androidx.core.net.toUri
 import com.example.zingmp3phake.R
+import com.example.zingmp3phake.data.model.Song
 import com.example.zingmp3phake.databinding.ActivityDetailPlaylistBinding
+import com.example.zingmp3phake.screen.detailsong.DetailSongActivity
+import com.example.zingmp3phake.screen.personal.RecyclerViewRecentAdapter
+import com.example.zingmp3phake.utils.ACTION_MUSIC
+import com.example.zingmp3phake.utils.ACTION_MUSIC_BROADCAST
+import com.example.zingmp3phake.utils.BUNDLE_LIST_KEY
+import com.example.zingmp3phake.utils.BUNDLE_TITLE_STRING_KEY
+import com.example.zingmp3phake.utils.DATA_KEY
+import com.example.zingmp3phake.utils.MEDIA_EXTERNAL_AUDIO_URI
+import com.example.zingmp3phake.utils.MusicAction
+import com.example.zingmp3phake.utils.SONG
 import com.example.zingmp3phake.utils.base.BaseActivity
+import com.example.zingmp3phake.utils.loadByGlide
+import kotlin.random.Random
 
 class DetailPlaylistActivity :
-    BaseActivity<ActivityDetailPlaylistBinding>(ActivityDetailPlaylistBinding::inflate) {
+    BaseActivity<ActivityDetailPlaylistBinding>(ActivityDetailPlaylistBinding::inflate),
+    RecyclerViewRecentAdapter.ItemClickListener,
+    DetailPlaylistContract.View {
 
+    private val adapterRV = RecyclerViewRecentAdapter(this)
+    private val detailPlaylistPresenter = DetailPlaylistPresenter(this)
     override fun initData() {
-        // TODO later
+        val mIntent = intent
+        val bundle = mIntent.getBundleExtra(DATA_KEY)
+        val title = bundle?.getString(BUNDLE_TITLE_STRING_KEY)
+        val list = bundle?.getParcelableArrayList<Song>(BUNDLE_LIST_KEY)
+        binding.textviewNumberOfListSong.text = "${list?.size} $SONG"
+        binding.recyclerView.adapter = adapterRV
+        if (list != null) adapterRV.setData(list)
+        if (title != null) binding.textviewPlaylistName.text = title
+
+        detailPlaylistPresenter.apply {
+            bindService(applicationContext)
+            val filter = IntentFilter(ACTION_MUSIC_BROADCAST)
+            registerReceiver(localReciver, filter)
+        }
     }
 
     override fun initView() {
-        setContentView(R.layout.activity_detail_playlist)
+        binding.apply {
+            buttonBack.setOnClickListener {
+                finish()
+            }
+            buttonNext.setOnClickListener {
+                getPendingIntent(MusicAction.NEXT.name)?.send()
+                binding.buttonPlay.setImageResource(R.drawable.ic_play_24)
+            }
+            buttonPlay.setOnClickListener {
+                getPendingIntent(MusicAction.PLAYORPAUSE.name)?.send()
+            }
+            buttonFavorite.setOnClickListener {
+                getPendingIntent(MusicAction.FAVORITE.name)?.send()
+            }
+            containerStateSong.setOnClickListener {
+                val intent = Intent(applicationContext, DetailSongActivity::class.java)
+                startActivity(intent)
+            }
```

Comment:
```suggestion
            buttonBack.setOnClickListener {
                finish()
            }
            buttonNext.setOnClickListener {
                getPendingIntent(MusicAction.NEXT.name)?.send()
                binding.buttonPlay.setImageResource(R.drawable.ic_play_24)
            }
            buttonPlay.setOnClickListener {
                getPendingIntent(MusicAction.PLAYORPAUSE.name)?.send()
            }
            buttonFavorite.setOnClickListener {
                getPendingIntent(MusicAction.FAVORITE.name)?.send()
            }
            containerStateSong.setOnClickListener {
                val intent = Intent(applicationContext, DetailSongActivity::class.java)
                startActivity(intent)
            }
```
Move hết đống này vào ``handleEvents`` em,
Nó không thuộc về ``initView``